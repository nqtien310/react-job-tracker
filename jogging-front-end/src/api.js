import Axios from 'axios'
import Auth from './auth'
export const API_ROOT = 'http://localhost:3001'

let axios = Axios.create({
  baseURL: API_ROOT
})

axios.interceptors.request.use(config => {
  if(Auth.getToken()){
    config.headers['Authorization'] = Auth.getToken()
  }
  return config
})

export const ADAPTER  = axios

class Api {
  constructor(adapter, apiRoot) {
    this.adapter = adapter
    this.root    = apiRoot
  }

  get(path, params={}) {
    return this.request("get", path, params)
  }

  post(path, params={}) {
    return this.request("post", path, params)
  }

  put(path, params={}) {
    return this.request("put", path, params)
  }

  delete(path, params={}) {
    return this.request("delete", path, params)
  }

  request(method, path, params={}){
    return this.adapter[method](path, params)
  }
}
export default (new Api(ADAPTER, API_ROOT))
