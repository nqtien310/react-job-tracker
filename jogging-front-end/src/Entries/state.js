import Rx from 'rxjs/Rx'
import api from '../api'
import { combineReducers } from 'redux'
import { push } from 'react-router-redux'
import setErrorMessage from '../actions/setErrorMessage.js'
import {reset} from 'redux-form'
//constants
export const ENTRY_FORM_CREATE = 'ENTRY_FORM_CREATE'
export const ENTRY_FORM_EDIT   = 'ENTRY_FORM_EDIT'
export const ENTRY_FORM_HIDE   = 'ENTRY_FORM_HIDE'
export const ENTRY_DELETE      = 'ENTRY_DELETE'
export const ENTRY_CREATE      = 'ENTRY_CREATE'
export const ENTRY_UPDATE      = 'ENTRY_UPDATE'
export const ENTRIES_FETCH     = 'ENTRIES_FETCH'
export const ENTRIES_FETCHED   = 'ENTRIES_FETCHED'

//actions
export const deleteEntry = (userId, entryId) => {
  return {
    type: ENTRY_DELETE,
    payload:{
      userId: userId,
      id: entryId
    }
  }
}

export const fetchEntries = (userId) => ({type: ENTRIES_FETCH, payload: userId})
export const showCreateForm = () => ({type: ENTRY_FORM_CREATE})
export const showEditForm = (entry) => ({type: ENTRY_FORM_EDIT, payload: entry})
export const hideForm = (entry) => ({type: ENTRY_FORM_HIDE})
export const createEntry = (userId, params) => {
  return {
    type: ENTRY_CREATE,
    payload: {
      userId: userId,
      params: params
    }
  }
}

export const updateEntry = (userId, id, params) => {
  return {
    type: ENTRY_UPDATE,
    payload: {
      userId: userId,
      id: id,
      params: params
    }
  }
}

export const fetchedEntries = (response) => {
  return {
    type: ENTRIES_FETCHED,
    payload: response.data
  }
}

//reducers
function listReducer(state=[], action){
  switch(action.type){
    case ENTRIES_FETCHED: {
      return action.payload
    }default: {
      return state
    }
  }
}

function formReducer(state={}, action){
  switch(action.type){
    case ENTRY_FORM_CREATE: {
      return {action: 'create'}
    }case ENTRY_FORM_EDIT: {
      return {action: 'edit', editing: action.payload}
    }case ENTRY_FORM_HIDE: {
      return {}
    }default: {
      return state
    }
  }
}

export const entryReducer = combineReducers({
  list: listReducer,
  form: formReducer
})

//Epics

export const deleteEntryEpic = (action$) => {
  return action$.ofType(ENTRY_DELETE).switchMap( action => {
    return Rx.Observable.fromPromise(
      api.delete(`users/${action.payload.userId}/entries/${action.payload.id}`)
    ).flatMap(response => [fetchEntries(action.payload.userId), hideForm()])
  })
}

export const updateEntryEpic = (action$) => {
  return action$.ofType(ENTRY_UPDATE).switchMap( action => {
    return Rx.Observable.fromPromise(
      api.put(`users/${action.payload.userId}/entries/${action.payload.id}`, action.payload.params)
    ).flatMap(response => [fetchEntries(action.payload.userId), hideForm()])
  })
}

export const fetchEntriesEpic = (action$) => {
  return action$.ofType(ENTRIES_FETCH).switchMap( action => {
    return Rx.Observable.fromPromise(
      api.get(`users/${action.payload}/entries`)
    ).map(response => fetchedEntries(response))
  })
}

export const createEntryEpic = (action$) => {
  return action$.ofType(ENTRY_CREATE).switchMap( action => {
    return Rx.Observable.fromPromise(
      api.post(`users/${action.payload.userId}/entries`, action.payload.params)
    ).flatMap(() => [fetchEntries(action.payload.userId), hideForm()]
    ).catch(error => Rx.Observable.of(setErrorMessage(error)))
  })
}


//https://github.com/redux-observable/redux-observable/tree/master/examples/navigation/epics
