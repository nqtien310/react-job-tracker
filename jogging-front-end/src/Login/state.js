import { fetchMyUser } from '../epics/fetchMyUserEpic'
import Auth from '../auth'
import 'rxjs/add/operator/switchMap';
import Rx from 'rxjs/Rx'
import api from '../api'
import setErrorMessage from '../actions/setErrorMessage.js'

export const LOGIN  = 'LOGIN'
export const SETTOKEN = 'SETTOKEN'

export const submitLogin = params => ({type: LOGIN, payload: params})
export const setToken    = params => ({type: SETTOKEN, payload: params})


export const loginEpic = (action$) => {
  return action$.ofType(LOGIN).switchMap(action =>{
    return Rx.Observable.fromPromise(
      api.post('login', {user: action.payload})
    ).flatMap(response =>
        [setToken(response), fetchMyUser()]
      ).catch(error => setErrorMessage(error))
  })
}

