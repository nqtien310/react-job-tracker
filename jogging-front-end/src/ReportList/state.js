import Auth from '../auth'
import api from '../api'
import Rx from 'rxjs/Rx'
import setErrorMessage from '../actions/setErrorMessage.js'
import { combineReducers } from 'redux'

export const REPORTS_FETCH = 'REPORTS_FETCH'
export const REPORTS_FETCHED = 'REPORTS_FETCHED'

export const fetchReports = (userId) => {
  return {
    type: REPORTS_FETCH,
    payload:{
      userId: userId
    }
  }
}

export const fetchedReports = (response) => {
  return {
    type: REPORTS_FETCHED,
    payload: response.data
  }
}

const listReducer = (state=[], action) => {
  switch(action.type){
    case REPORTS_FETCHED: {
      return action.payload
    }default: {
      return state
    }
  }
}

export const reportReducer = combineReducers({
  list: listReducer
})

export const fetchReportsEpic = (action$) => {
  return action$.ofType(REPORTS_FETCH).switchMap((action) => {
    return Rx.Observable.fromPromise(
      api.get(`/users/${action.payload.userId}/weekly_summaries`)
    ).flatMap(response => [fetchedReports(response)])
  })
}
