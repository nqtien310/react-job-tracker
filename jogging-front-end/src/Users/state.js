import { combineReducers } from 'redux'
import { push } from 'react-router-redux'
import Rx from 'rxjs/Rx'
import api from '../api'
import setErrorMessage from '../actions/setErrorMessage'

export const USERS_FETCH = 'USERS_FETCH'
export const USER_FETCH = 'USER_FETCH'

export const USER_FETCHED = 'USER_FETCHED'
export const USERS_FETCHED = 'USERS_FETCHED'
export const USER_CREATE  = 'SUBMIT_CREATE_USER'

export const USER_UPDATE = 'USER_UPDATE'

export const fetchUsers = () => ({type: USERS_FETCH})
export const fetchUser = (userId) => ({type: USER_FETCH, payload: userId})
export const fetchedUser = (response) => ({type: USER_FETCHED, payload: response.data})
export const fetchedUsers = (response) => ({type: USERS_FETCHED, payload: response.data})
export const createUser = params => ({type: USER_CREATE, payload: params})
export const updateUser = (userId, params) => ({type: USER_UPDATE, payload: {params: params, userId: userId}})

function listReducer(state=[], action){
  switch(action.type) {
    case USERS_FETCHED: {
      return action.payload
    }default: {
      return state
    }
  }
}

function selectedReducer(state=null, action){
  switch(action.type) {
    case USER_FETCHED: {
      return action.payload
    }default: {
      return state
    }
  }
}

export const userReducer = combineReducers({
  list: listReducer,
  selected: selectedReducer
})

export const fetchUsersEpic = (action$) => {
  return action$.ofType(USERS_FETCH).switchMap((action) => {
    return Rx.Observable.fromPromise(api.get("/users"))
      .flatMap(response => [fetchedUsers(response)])
  })
}

export const fetchUserEpic = (action$) => {
  return action$.ofType(USER_FETCH).switchMap((action) => {
    return Rx.Observable.fromPromise(api.get(`/users/${action.payload}`))
      .flatMap(response => [fetchedUser(response)])
  })
}

export const createUserEpic = (action$) => {
  return action$.ofType(USER_CREATE).switchMap(action =>{
    return Rx.Observable.fromPromise(
      api.post('/register', {user: action.payload})
    ).flatMap(response => [push('/login'), setErrorMessage(null)]
    ).catch(error => Rx.Observable.of(setErrorMessage(error)))
  })
}

export const updateUserEpic = (action$) => {
  return action$.ofType(USER_UPDATE).switchMap(action =>{
    return Rx.Observable.fromPromise(
      api.put(`/users/${action.payload.userId}`, {user: action.payload.params})
    ).flatMap(response => [push('/'), setErrorMessage(null)]
    ).catch(error => Rx.Observable.of(setErrorMessage(error)))
  })
}

