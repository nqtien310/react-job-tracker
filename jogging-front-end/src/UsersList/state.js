import { combineReducers } from 'redux'
import Rx from 'rxjs/Rx'
import api from '../api'

export const USERS_FETCH = 'USERS_FETCH'
export const USERS_FETCHED = 'USERS_FETCHED'

export const fetchUsers = () => ({type: USERS_FETCH})
export const fetchedUsers = (response) => ({type: USERS_FETCHED, payload: response.data})

function listReducer(state=[], action){
  switch(action.type) {
    case USERS_FETCHED: {
      return action.payload
    }default: {
      return state
    }
  }
}

export const userReducer = combineReducers({
  list: listReducer
})

export const fetchUsersEpic = (action$) => {
  return action$.ofType(USERS_FETCH).switchMap((action) => {
    return Rx.Observable.fromPromise(api.get("/users"))
      .flatMap(response => [fetchedUsers(response)])
  })
}
