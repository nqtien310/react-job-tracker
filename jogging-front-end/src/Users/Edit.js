import React from 'react';
import Template from '../Template'
import Form from './Form'
import { fetchUser, updateUser } from './state'
import { connect } from 'react-redux'

class EditForm extends React.Component {
  constructor(props){
    super(props)
    this.userId = this.props.match.params.userId
  }

  componentWillMount() {
    this.props.fetchUser(this.userId)
  }

  onSubmit = (params) => {
    this.props.updateUser(this.userId, params)
  }

  render() {
    return (
      <Form onSubmit={this.onSubmit} initialValues={this.props.selectedUser}/>
    )
  }
}

function mapStateToProps(state){
  return {
    selectedUser: state.user.selected
  }
}

export default connect(mapStateToProps, {
  fetchUser,
  updateUser
})(EditForm)
