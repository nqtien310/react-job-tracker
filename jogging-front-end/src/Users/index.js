import React from 'react';
import { connect } from 'react-redux'
import { fetchUsers } from './state'
import { Link } from 'react-router-dom'

class UsersList extends React.Component{
  componentWillMount(){
    this.props.fetchUsers()
  }

  onDelete() {
    console.log("DELETE")
  }

  onEdit() {
    console.log("EDIT")
  }

  renderUsers() {
    return this.props.users.map(user => {
      return(
        <tr key={user.id}>
          <td> {user.email} </td>
          <td> {user.full_name} </td>
          <td className="actions">
            <a onClick={()=> this.onDelete(user.id)} className="btn btn-danger">Delete</a>
            <Link to={`/users/${user.id}/edit`} className="btn btn-warning">Edit</Link>
            <Link to={`/users/${user.id}/entries`} className="btn btn-primary">Entries</Link>
            <Link to={`/users/${user.id}/report`} className="btn btn-primary">Report</Link>
          </td>
        </tr>
      )
    })
  }

  render() {
    return (
      <table className="table">
        <thead>
          <tr>
            <th>Email</th>
            <th>Full Name</th>
            <th>Action</th>
          </tr>
        </thead>

        <tbody>
          {this.renderUsers()}
        </tbody>
      </table>
    )
  }
}

function mapStateToProps(state){
  return {
    users: state.user.list
  }
}
export default connect(mapStateToProps, {
  fetchUsers
})(UsersList)
