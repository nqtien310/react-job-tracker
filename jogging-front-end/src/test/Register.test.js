import React from 'react';
import renderer from 'react-test-renderer'
import RenderComponent, {store} from './RenderComponent'
import RegisterForm from '../Register'
import { Api } from '../api'
import { createUser } from '../Users/state'
import {stubApiError} from './ApiStubs'

it('Register Form', () => {
  const component = renderer.create(
    <RenderComponent>
      <RegisterForm/>
    </RenderComponent>
  )

  expect(component).toMatchSnapshot()
});

it('displays error when register failed', (done) => {
  stubApiError("post", {"message" : "Email has already been taken"})
  store.dispatch(createUser({}))

  setTimeout( () => {
    let component = renderer.create(
      <RenderComponent>
        <RegisterForm/>
      </RenderComponent>
    )
    expect(component).toMatchSnapshot()
    done()
  }, 50)
})
