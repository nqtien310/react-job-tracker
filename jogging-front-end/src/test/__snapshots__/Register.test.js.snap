// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Form 1`] = `
<div>
  <nav
    className="navbar navbar-toggleable-md navbar-inverse bg-inverse"
  >
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler navbar-toggler-right"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <a
      className="navbar-brand"
      href="#"
    >
      <i
        className="fa fa-home"
      />
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item active"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    id="body"
  >
    <div
      id="user-form"
    >
      <form
        onSubmit={[Function]}
      >
        <div />
        <div>
           
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="email"
            >
              Email
            </label>
            <div
              className="col-sm-9"
            >
              <input
                autoFocus={true}
                className="form-control"
                id="email"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="full_name"
            >
              Full Name
            </label>
            <div
              className="col-sm-9"
            >
              <input
                autoFocus={false}
                className="form-control"
                id="full_name"
                name="full_name"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="password"
            >
              Password
            </label>
            <div
              className="col-sm-9"
            >
              <input
                autoFocus={false}
                className="form-control"
                id="password"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="password_confirmation"
            >
              Confirm
            </label>
            <div
              className="col-sm-9"
            >
              <input
                autoFocus={false}
                className="form-control"
                id="password_confirmation"
                name="password_confirmation"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
            >
              Role
            </label>
            <div
              className="col-sm-9 radio-btns-group"
            >
              <label>
                <input
                  checked={false}
                  name="role"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="radio"
                  value="admin"
                />
                 
                admin
              </label>
              <label>
                <input
                  checked={false}
                  name="role"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="radio"
                  value="manager"
                />
                 
                manager
              </label>
              <label>
                <input
                  checked={false}
                  name="role"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="radio"
                  value="user"
                />
                 
                user
              </label>
            </div>
          </div>
           
        </div>
        <div
          className="form-group row"
        >
          <div
            className="offset-sm-3 col-sm-9"
          >
            <button
              className="btn btn-primary"
              type="Submit"
            >
               
              Register
               
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`displays error when register failed 1`] = `
<div>
  <nav
    className="navbar navbar-toggleable-md navbar-inverse bg-inverse"
  >
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler navbar-toggler-right"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <a
      className="navbar-brand"
      href="#"
    >
      <i
        className="fa fa-home"
      />
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item active"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    id="body"
  >
    <div
      id="user-form"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="alert alert-danger"
        >
          <ul>
            <li>
               
              Email has already been taken
               
            </li>
          </ul>
        </div>
        <div>
           
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="email"
            >
              Email
            </label>
            <div
              className="col-sm-9"
            >
              <input
                autoFocus={true}
                className="form-control"
                id="email"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="full_name"
            >
              Full Name
            </label>
            <div
              className="col-sm-9"
            >
              <input
                autoFocus={false}
                className="form-control"
                id="full_name"
                name="full_name"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="password"
            >
              Password
            </label>
            <div
              className="col-sm-9"
            >
              <input
                autoFocus={false}
                className="form-control"
                id="password"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="password_confirmation"
            >
              Confirm
            </label>
            <div
              className="col-sm-9"
            >
              <input
                autoFocus={false}
                className="form-control"
                id="password_confirmation"
                name="password_confirmation"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
            >
              Role
            </label>
            <div
              className="col-sm-9 radio-btns-group"
            >
              <label>
                <input
                  checked={false}
                  name="role"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="radio"
                  value="admin"
                />
                 
                admin
              </label>
              <label>
                <input
                  checked={false}
                  name="role"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="radio"
                  value="manager"
                />
                 
                manager
              </label>
              <label>
                <input
                  checked={false}
                  name="role"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="radio"
                  value="user"
                />
                 
                user
              </label>
            </div>
          </div>
           
        </div>
        <div
          className="form-group row"
        >
          <div
            className="offset-sm-3 col-sm-9"
          >
            <button
              className="btn btn-primary"
              type="Submit"
            >
               
              Register
               
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
